tests/lib/fakeinstaller/main.go

```diff
@@ -158,28 +208,40 @@ func createAndMountFilesystems(bootDevice string, volumes map[string]*gadget.Vol
 	vol := firstVol(volumes)
 
 	var mountPoints []string
-	for _, stru := range vol.Structure {
-		if stru.Label == "" || stru.Filesystem == "" {
+	for _, volStruct := range vol.Structure {
+		if volStruct.Label == "" || volStruct.Filesystem == "" {
 			continue
 		}
 
-		part, err := disk.FindMatchingPartitionWithPartLabel(stru.Label)
-		if err != nil {
-			return nil, err
+		var partNode string
+		if shouldEncrypt && (volStruct.Role == gadget.SystemSave || volStruct.Role == gadget.SystemData) {
+			encryptedDevice := encryptedDevices[volStruct.Role]
+			if encryptedDevice == "" {
+				return nil, fmt.Errorf("no encrypted device found for %s role", volStruct.Role)
+			}
+			partNode = encryptedDevice
+		} else {
+			part, err := disk.FindMatchingPartitionWithPartLabel(volStruct.Label)
+			if err != nil {
+				return nil, err
+			}
+			partNode = part.KernelDeviceNode
 		}
-		if err := mkfs.Make(stru.Filesystem, part.KernelDeviceNode, stru.Label, 0, 0); err != nil {
+
+		logger.Debugf("making filesystem in %q", partNode)
+		if err := mkfs.Make(volStruct.Filesystem, partNode, volStruct.Label, 0, 0); err != nil {
```

Comment:
so here we are passing 0 0 atm, but here it's where we the sector size if we cared would play a role